beetsplug/parentwork.py

```diff
@@ -0,0 +1,159 @@
+        cmd.func = self.command
+        return [cmd]
+
+    def command(self, lib, opts, args):
+        self.find_work(lib.items(ui.decargs(args)))
+
+    item_types = {
+        'parent_work':          types.STRING,
+        'parent_work_disambig': types.STRING,
+        'work':                 types.STRING,
+        'work_disambig':        types.STRING,
+        'performer':            types.STRING,
+        'performer_sort':       types.STRING,
+        'parent_composer':      types.STRING,
+        'parent_composer_sort': types.STRING}
+
+    def imported(self, session, task):
+        self.find_work(task.items)
+
+    def find_work(self, items):
+    
+        def find_parentwork(work_id):
+            "finds the parentwork of a work given its id"
+            work_info = musicbrainzngs.get_work_by_id(work_id,
+                includes=["work-rels", "artist-rels"])
+            partof = True
+            while partof:
+                partof = False
+                if 'work-relation-list' in work_info['work']:
+                    for work_father in work_info['work'][
+                            'work-relation-list']:
+                        if work_father['type'] == 'parts' and\
+                                'direction' in work_father:
+                            if work_father['direction'] == 'backward':
+                                father_id = work_father['work']['id']
+                                partof = True
+                                work_info = musicbrainzngs.get_work_by_id(
+                                    father_id,includes=["work-rels", 
+                                        "artist-rels"])
+            return work_info
+
+        for item in items:
+            performer            = []
+            performer_sort       = []
+            work                 = []
+            work_disambig        = []
+            parent_work          = []
+            parent_work_disambig = []
+            parent_composer      = []
+            parent_composer_sort = []
+            item.read()
+            recording_id = item['mb_trackid']
+            performer_types = ['performer', 'instrument', 'vocal',
+                'conductor', 'performing orchestra', 'chorus master', 
+                    'concertmaster']
+            i = 0
+            while i < 5:
+                try:
+                    rec_rels = musicbrainzngs.get_recording_by_id(
+                        recording_id, includes=['work-rels', 'artist-rels'])
+                    if 'artist-relation-list' in rec_rels['recording']:
+                        for dudes in rec_rels['recording'][
+                                'artist-relation-list']:
+                            if dudes['type'] in performer_types:
+                                performer.append(dudes['artist']['name'])
+                                    performer_sort.append(dudes['artist']
+                                        ['sort-name'])
```

Comment:
Yes, it takes performers of the current recording to be exact. After input of bearcatsandor I would agree and put it into the main code instead of this plugin. Maybe we should not only catch the name of the performers but also their roles (instrument-which one?/orchestra/voice/DJ etc)